crates/bevy_core_pipeline/src/bloom/mod.rs

```diff
@@ -576,180 +370,122 @@ fn prepare_bloom_textures(
                 sample_count: 1,
                 dimension: TextureDimension::D2,
-                format: ViewTarget::TEXTURE_FORMAT_HDR,
+                format: BLOOM_TEXTURE_FORMAT,
                 usage: TextureUsages::RENDER_ATTACHMENT | TextureUsages::TEXTURE_BINDING,
                 view_formats: &[],
             };
 
-            texture_descriptor.label = Some("bloom_texture_a");
-            let texture_a = texture_as
-                .entry(camera.target.clone())
-                .or_insert_with(|| texture_cache.get(&render_device, texture_descriptor.clone()))
-                .clone();
-
-            texture_descriptor.label = Some("bloom_texture_b");
-            let texture_b = texture_bs
-                .entry(camera.target.clone())
-                .or_insert_with(|| texture_cache.get(&render_device, texture_descriptor))
-                .clone();
-
-            commands.entity(entity).insert(BloomTextures {
-                texture_a,
-                texture_b,
+            commands.entity(entity).insert(BloomTexture {
+                texture: texture_cache.get(&render_device, texture_descriptor),
                 mip_count,
             });
         }
     }
 }
 
-/// The uniform struct extracted from [`BloomSettings`] attached to a [`Camera`].
-/// Will be available for use in the Bloom shader.
-#[doc(hidden)]
-#[derive(Component, ShaderType, Clone)]
-pub struct BloomUniform {
-    threshold: f32,
-    knee: f32,
-    scale: f32,
-    intensity: f32,
-    viewport: Vec4,
-}
-
 #[derive(Component)]
 struct BloomBindGroups {
-    prefilter_bind_group: BindGroup,
     downsampling_bind_groups: Box<[BindGroup]>,
     upsampling_bind_groups: Box<[BindGroup]>,
-    upsampling_final_bind_group: BindGroup,
+    sampler: Sampler,
 }
 
 fn queue_bloom_bind_groups(
     mut commands: Commands,
     render_device: Res<RenderDevice>,
-    pipelines: Res<BloomPipelines>,
-    uniforms: Res<ComponentUniforms<BloomUniform>>,
-    views: Query<(Entity, &ViewTarget, &BloomTextures)>,
+    downsampling_pipeline: Res<BloomDownsamplingPipeline>,
+    upsampling_pipeline: Res<BloomUpsamplingPipeline>,
+    views: Query<(Entity, &BloomTexture)>,
 ) {
-    if let Some(uniforms) = uniforms.binding() {
-        for (entity, view_target, textures) in &views {
-            let prefilter_bind_group = render_device.create_bind_group(&BindGroupDescriptor {
-                label: Some("bloom_prefilter_bind_group"),
-                layout: &pipelines.downsampling_bind_group_layout,
+    for (entity, bloom_texture) in &views {
+        let bind_group_count = bloom_texture.mip_count as usize - 1;
+
+        let sampler = render_device.create_sampler(&SamplerDescriptor {
+            min_filter: FilterMode::Linear,
+            mag_filter: FilterMode::Linear,
+            address_mode_u: AddressMode::ClampToEdge,
+            address_mode_v: AddressMode::ClampToEdge,
+            ..Default::default()
+        });
```

Comment:
We probably shouldn't create a new sampler every frame. We could probably create one sampler and put it in a resource, if there isn't a better way.

Reply:
We should probably have a sampler resource each pass can use instead of making their own. I'll open an issue.